app/services/solr_mapper.rb

```diff
@@ -14,15 +14,90 @@ def initialize(metadata:, dataset_record_id:, dataset_record_set_id:)
   end
 
   # @return [Hash] the Solr document
+  # rubocop:disable Metrics/AbcSize
   def call
     {
       id: dataset_record_id,
-      dataset_record_set_id: dataset_record_set_id,
-      title: metadata.dig(:titles, 0, :title)
-    }
+      dataset_record_set_id_ss: dataset_record_set_id,
+      access_ssi: metadata['access'],
+      provider_ssi: metadata['provider'],
+      creators_struct_ss: metadata['creators'].to_json,
+      descriptions_tsim: retrieve_descriptions(metadata['descriptions'] || []),
+      doi_ssi: retrieve_dois(metadata['identifiers']),
+      provider_identifier_ssim: map_provider_identifiers(metadata['identifiers'], metadata['provider'])
+    }.merge(map_titles(metadata['titles']))
   end
+  # rubocop:enable Metrics/AbcSize
 
   private
 
   attr_reader :metadata, :dataset_record_id, :dataset_record_set_id
+
+  # Given titles from metadata, return field value based on title type
+  def title_fields(title_type, titles)
+    titles.select { |title_obj| title_obj.key?('title_type') && title_obj['title_type'] == title_type }.pluck('title')
```

Comment:
Why do you need to check if the key exists?

Reply:
In the example from the dataworks schema validator, the title_type key doesn't always exist. https://github.com/sul-dlss/dataworks-etl/blob/main/spec/services/dataworks_validator_spec.rb#L56

Reply:
If title_type doesn't exist then `title_obj['title_type']` will return `nil` which won't match `title_type`.

Reply:
Thanks. Changed.